servers/zms/src/main/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnection.java

```diff
@@ -3441,6 +3441,72 @@ public boolean confirmRoleMember(String domainName, String roleName, RoleMember
         return result;
     }
 
+    @Override
+    public Map<String, List<DomainRoleMember>> getPendingDomainRoleMembersList(String principal) {
+
+        final String caller = "getPendingDomainRoleMembersList";
+        int principalId = getPrincipalId(principal);
+        if (principalId == 0) {
+            throw notFoundError(caller, ZMSConsts.OBJECT_PRINCIPAL, principal);
+        }
+        int auditDomId = getDomainId(SYS_AUTH_AUDIT_DOMAIN);
+
+        Map<String, List<DomainRoleMember>> domainRoleMembersMap = new LinkedHashMap<>();
+        String domain;
+        List<DomainRoleMember> domainRoleMembers;
+        DomainRoleMember domainRoleMember;
+        List<MemberRole> memberRoles;
+        MemberRole memberRole;
+
+        try (PreparedStatement ps = con.prepareStatement(SQL_PENDING_DOMAIN_ROLE_MEMBER_LIST)) {
+            ps.setInt(1, principalId);
+            ps.setInt(2, auditDomId);
+            try (ResultSet rs = executeQuery(ps, caller)) {
+                while (rs.next()) {
+                    domain = rs.getString(1);
+                    domainRoleMembers = domainRoleMembersMap.get(domain);
+                    if (domainRoleMembers != null) {
```

Comment:
there seems to be a lot of code repetition here. you can use the following pattern:
if (!domainRoleMembersMap.contains(domain)) {
    // create and insert a new List
}

then just get the list and add a new member to it.